utils.js

```diff
@@ -1,263 +1,295 @@
-
-  const JSZIP = require('jszip');
-  const zip = new JSZIP();
-
-  //Get the contents of the zip buffer to get the details
-  const zipFile = zip.loadAsync(details.content);
-
-  return zipFile.then(zip => {
-    //Filter out hidden files. Coveo can't really extract anything from these and some users
-    //may prefer not to have their hidden files pushed into an index.
-    let names = Object.keys(zip.files).filter(file => !/(^|\/)\.[^/.]/g.test(file));
-
-    //Track folder names and number of files currently prepped for upload. Also
-    //track total file size so the user doesn't send files that are too big.
-    let folderNames = [];
-    let fileCount = 0;
-    let totalFileSize = 0;
-
-    //Iterate through each file in the zip
-    names.forEach(name => {
-      let zipContent = {};
-
-      // These files are macOS dependent and don't have any valuable content for Coveo to extract, so ignore them.
-      if (name.indexOf('__MACOSX/') > -1) {
-        // Also ignore folders within the zip file, just want their contents not the folders themselves, but keep folder names
-        // to help alter the file names within it.
-      } else if (zip.files[name].dir) {
-        folderNames.unshift(name);
-      } else {
-        //Extract important details for the files here
-        zipContent.size = zip.files[name]._data.compressedSize;
-        totalFileSize += zipContent.size;
-        zipContent.contentType = path.extname(name);
-        zipContent.filename = name;
-
-        //The files are too big (100 MB for now), throw an error telling
-        //the user so and the file size limit.
-        if (totalFileSize >= 100 * 1024 * 1024) {
-          throw new Error(messages.BIG_FILE);
-        }
-
-        zipContent.content = zip.files[name]._data.compressedContent;
-        fileCount++;
-
-        //Set limit on number of documents that a zip file can contain in order to avoid
-        //customers sending tons of files at once. Currently set at 50.
-        if (fileCount > 50) {
-          throw new Error(messages.TOO_MANY_FILES);
-        }
-
-        //If a file is in a folder, the file name includes the folder's name
-        //as well. This looks at the encountered folders so far and removes
-        //the excess folder name out of the file name.
-        folderNames.forEach(folderName => {
-          if (zipContent.filename.indexOf(folderName) > -1) {
-            zipContent.filename = name.substr(folderName.length, name.length);
-          }
-        });
-
-        //Get the compression type of the individual file in the zip file
-        zipContent.compressionType = findCompressionType(zipContent, zip.files[name]._data.uncompressedSize);
-
-        //Push the file information into the array to be handled later
-        addOrUpdate.push(zipContent);
-      }
-    });
 
-    return addOrUpdate;
-  });
-};
+//Function to detect if the file being observed is a supported archive
+//filetype or not.
+const archiveTypeChecker = details => {
```

Comment:
Should rename this function to something like `validateArchiveType` to be more explicit about what it does